test/meta_testing_tools.jl

```diff
@@ -107,8 +107,13 @@ end
                 end
             end
             @test length(fails) === 2
-            @test fails[1].orig_expr == :(false==true)
-            @test fails[2].orig_expr == :(true==false)
+            
+
+            # Newer versions of Julia return a `String`, not an `Expr`. 
+            # Always calling  `string` on it gives gives consistency regardless of version.
+            # https://github.com/JuliaLang/julia/pull/37809
+            @test string(fails[1].orig_expr) == "false == true"
+            @test string(fails[2].orig_expr) == "true == false"
```

Comment:
alternatively we could check if `orig_expr` is a `String` (or check the Julia version) and `Meta.parse` it to an `Expr`?

Reply:
then we'd be testing what we want to test (which is that the expression is the one we expect), rather than testing the string representation

Reply:
indeed, but it is more complex

Reply:
I still lean towards simpler is better

Reply:
I've initially done what @nickrobinson251 suggested but I prefer this. In particular the current one which stringifies the expression